cinema/views.py

```diff
@@ -101,8 +102,25 @@ def get_serializer_class(self):
         if self.action == "retrieve":
             return MovieDetailSerializer
 
+        if self.action == "upload_image":
+            return MovieImageSerializer
```

Comment:
The `upload_image` action in `MovieViewSet` correctly uses `MovieImageSerializer` to handle image uploads, ensuring that the image URL is updated in the movie instance.